alpa/pipeline_parallel/cross_mesh_resharding.py

```diff
@@ -195,6 +195,8 @@ def __init__(self, task_spec, collective_group, src_mesh, dst_mesh):
         self.send_worker_task_ids = {}
         self.recv_worker_task_ids = {}
 
+        self.task_dones = []
```

Comment:
remove unused code. For a series of tasks, we always create a tmp task dones instead of keeping using the same one